### PR TITLE
Update readme for bktec

### PR DESCRIPTION
**Pull request by amybiyuliu**
### Description

The README for bktec v.1.0.0 release. 

- Updates README to reflect the new name Test Engine Client (bktec)
- Add docs for Jest

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,87 +1,87 @@
-# Buildkite Test Splitter
+# Buildkite Test Engine Client
 
-Buildkite Test Splitter is an open source tool to orchestrate your test suites. It uses your Buildkite Test Analytics suite data to intelligently partition and parallelise your tests. 
-Test Splitter currently only supports RSpec but support for other frameworks is coming soon.
+Buildkite Test Engine Client (bktec) is an open source tool to orchestrate your test suites. It uses your Buildkite Test Engine suite data to intelligently partition and parallelise your tests.
+
+bktec supports RSpec and Jest.
 
 ## Migrating to 0.9.0
```
Comment (amybiyuliu) [inline README.md:7]: Do we need to keep this part for v1.0.0?

File: README.md [hunk 2]
```diff
@@ -1,87 +1,82 @@
...
-| `BUILDKITE_SPLITTER_RESULT_PATH` | Test Splitter uses this environment variable to tell the runner where to store the test result. Test Splitter reads the test result after each test run for retries and verification. For RSpec, the result is generated using the `--format json` and `--out` CLI options, while for Jest, it is generated using the `--json` and `--outputFile` options. We have included these options in the default test command for RSpec and Jest. If you need to customize your test command, make sure to append the CLI options to save the result to a file. Please refer to the `BUILDKITE_SPLITTER_TEST_CMD` environment variable for more details. <br> *Note: Test Splitter will not delete the file after running the test, however it will be deleted by Buildkite Agent as part of build lifecycle. *|
-
+| `BUILDKITE_TEST_ENGINE_API_ACCESS_TOKEN ` | Buildkite API access token with `read_suites`, `read_test_plan`, and `write_test_plan` scopes. You can create an access token from [Personal Settings](https://buildkite.com/user/api-access-tokens) in Buildkite |
```
Comment (amybiyuliu) [inline README.md:38]: Do we need a migrating to 1.0.0 section as we renamed a lot envs and they will be breaking changes

Comment (nprizal) [reply README.md:38]: Yes, I think we need to add the section.

File: README.md [hunk 3]
```diff
@@ -1,87 +1,99 @@
...
-Test Splitter currently only supports RSpec but support for other frameworks is coming soon.
+Buildkite Test Engine Client (bktec) is an open source tool to orchestrate your test suites. It uses your Buildkite Test Engine suite data to intelligently partition and parallelise your tests.
 
-## Migrating to 0.9.0
-Version 0.9.0 introduces a new feature that requires Test Splitter to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_SPLITTER_RESULT_PATH` environment variable. This variable specifies the location of where the runner should store test results.
+bktec supports RSpec and Jest.
 
-Furthermore, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_SPLITTER_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command. 
+## Migrating to 1.0.0
```
Comment (niceking) [inline README.md:7]: Is `BUILDKITE_TEST_ENGINE_RESULT_PATH` a new mandatory env var in v1.0.0?

File: README.md [hunk 4]
```diff
@@ -1,87 +1,99 @@
...
+| `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN` | For RSpec:<br> -<br> For Jest:<br> `node_modules` | Glob pattern to exclude certain test files or directories. The exclusion will be applied after discovering the test files using a pattern configured with `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN`. </br> *This option accepts the pattern syntax supported by the [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library.* |
+| `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN` | For Rspec:</br> `spec/**/*_spec.rb`</br>  For Jest:</br> `**/{__tests__/**/*,*.spec,*.test}.{ts,js,tsx,jsx}` | Glob pattern to discover test files. You can exclude certain test files or directories from the discovered test files using a pattern that can be configured with `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN`.</br> *This option accepts the pattern syntax supported by the [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library.* |
+| `BUILDKITE_TEST_ENGINE_TEST_RUNNER` | `rspec` | Test runner to use for running tests. Currently `rspec` and `jest` are supported.
 
 
-### Run the Test Splitter
+### Runnnig bktec
```
Comment (niceking) [inline README.md:74]:
```suggestion
### Running bktec
```

File: README.md [hunk 5]
```diff
@@ -1,87 +1,99 @@
...
+| `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN` | For RSpec:<br> -<br> For Jest:<br> `node_modules` | Glob pattern to exclude certain test files or directories. The exclusion will be applied after discovering the test files using a pattern configured with `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN`. </br> *This option accepts the pattern syntax supported by the [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library.* |
+| `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN` | For Rspec:</br> `spec/**/*_spec.rb`</br>  For Jest:</br> `**/{__tests__/**/*,*.spec,*.test}.{ts,js,tsx,jsx}` | Glob pattern to discover test files. You can exclude certain test files or directories from the discovered test files using a pattern that can be configured with `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN`.</br> *This option accepts the pattern syntax supported by the [zzglob](https://github.com/DrJosh9000/zzglob?tab=readme-ov-file#pattern-syntax) library.* |
+| `BUILDKITE_TEST_ENGINE_TEST_RUNNER` | `rspec` | Test runner to use for running tests. Currently `rspec` and `jest` are supported.
```
Comment (niceking) [inline README.md:71]: I'm a little surprised that we're setting `rspec` as a default value 🤔 is this because of legacy reasons?

**Review by niceking — COMMENTED** [review]
TC are going to upgrade to v1.0.0.rc1 (or whatever we're up to) so let's merge this so they can follow the upgrade instructions

File: README.md (hunk 3 above)
Comment (amybiyuliu) [reply README.md:7]: It became mandatory from 0.9.0
Do we want to highlight it's mandatory in here as well?

File: README.md (hunk 5 above)
Comment (amybiyuliu) [reply README.md:71]: I think so. We made `rspec` as default while preparing the Jest support if I remember it correct.

File: README.md (hunk 3 above)
Comment (nprizal) [reply README.md:7]: I think so, If we remove the "Migrating to 0.9.0", some customer could miss that update.

File: README.md (hunk 5 above)
Comment (nprizal) [reply README.md:71]: Yeah, I think we didn't want to break the old version. Maybe it's the time to make it mandatory (no default)?

Comment (niceking) [reply README.md:71]: Yeah that's a good suggestion @nprizal

Comment (amybiyuliu) [reply README.md:71]: I'll make a ticket for making test runner mandatory and working on it

File: README.md [hunk 6]
```diff
@@ -1,87 +1,106 @@
...
-Test Splitter currently only supports RSpec but support for other frameworks is coming soon.
+Buildkite Test Engine Client (bktec) is an open source tool to orchestrate your test suites. It uses your Buildkite Test Engine suite data to intelligently partition and parallelise your tests.
+
+bktec supports RSpec and Jest.
 
 ## Migrating to 0.9.0
-Version 0.9.0 introduces a new feature that requires Test Splitter to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_SPLITTER_RESULT_PATH` environment variable. This variable specifies the location of where the runner should store test results.
+Version 0.9.0 introduces a new feature that requires bktec to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_SPLITTER_RESULT_PATH` environment variable. This variable specifies the location of where the runner should store test results.
```
Comment (nprizal) [inline README.md:8]: I think we can merge this to migrating to `1.0.0` section as the env var name is now also changed.

File: test-engine-client
Comment (nprizal) [inline test-engine-client:1]: I think this is accidentally included in the commit. Can we remove it?

File: README.md [hunk 7]
```diff
@@ -1,87 +1,105 @@
...
 
-## Migrating to 0.9.0
-Version 0.9.0 introduces a new feature that requires Test Splitter to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_SPLITTER_RESULT_PATH` environment variable. This variable specifies the location of where the runner should store test results.
+bktec supports RSpec and Jest.
 
-Furthermore, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_SPLITTER_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command. 
+## Migrating to 1.0.0
+Version 0.9.0 introduces a new feature that requires bktec to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_TEST_ENGINE_RESULT_PATH` environment variable. This variable specifies the location of where the runner should store test results.
```
Comment (nprizal) [inline README.md:8]: Hmm now that I see the changes, I don't think this make sense. How about grouping the new manadatory env variables for the `1.0.0`?
```
Following environment variables is now required
- `BUILDKITE_TEST_ENGINE_RUNNER`
    {{description}}
- `BUILDKITE_TEST_ENGINE_RESULT_PATH`
   {{description}}

In addition, version 1.0.0 introduces name changes to environment variables....
| Old variable | New variable |
....

@wooly what do you think?

**Comment by ags** [comment]
> TC are going to upgrade to v1.0.0.rc1 (or whatever we're up to) so let's merge this so they can follow the upgrade instructions

As long as TC is "The Conversation", we've upgraded to v1.0.0.rc3 on a branch now 😅

File: README.md (hunk 7 above)
Comment (ags) [reply README.md:8]: This would be a bit nicer!

File: README.md (hunk 3 above)
Comment (ags) [reply README.md:7]: It'd be great if `BUILDKITE_TEST_ENGINE_RESULT_PATH` could default to a tempfile or something 🤔

File: README.md (hunk 7 above)
Comment (amybiyuliu) [reply README.md:8]: Thanks for the feedback @ags I'll group the required variables for better readability. Also great to hear you upgraded to v1.0.0-rc3

File: README.md (hunk 3 above)
Comment (nprizal) [reply README.md:7]: Thanks for the feedback. I agree, I think we can make the `BUILDKITE_TEST_ENGINE_RESULT_PATH` optional and default to a temp file. However, you still need to specify the `--format json --out {{resultPath}}` when using custom `BUILDKITE_TEST_ENGINE_TEST_CMD`.

File: README.md [hunk 8]
```diff
@@ -1,87 +1,109 @@
...
+
+## Migrating to 1.0.0
+
+The following environment variables are now required
+- `BUILDKITE_TEST_ENGINE_RUNNER`
+  The test runner to use for running tests. You will need to ensure that `BUILDKITE_TEST_ENGINE_TEST_RUNNER` presents in the environment. Currently `rspec` and `jest` are supported.
+- `BUILDKITE_TEST_ENGINE_RESULT_PATH`
+  The location of where the runner should store test results. We introcuded a new feature that requires bktec to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_TEST_ENGINE_RESULT_PATH` environment variable.
+
+In addition, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_TEST_ENGINE_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command.
```
Comment (nprizal) [inline README.md:15]:
```suggestion
- `BUILDKITE_TEST_ENGINE_RUNNER`
 
  The test runner to use for running tests. You will need to ensure that `BUILDKITE_TEST_ENGINE_TEST_RUNNER` presents in the environment. Currently `rspec` and `jest` are supported.
  
- `BUILDKITE_TEST_ENGINE_RESULT_PATH`
  
  The location of where the runner should store test results. We introcuded a new feature that requires bktec to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_TEST_ENGINE_RESULT_PATH` environment variable.

  In addition, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_TEST_ENGINE_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command. 
```

Comment (wooly) [reply README.md]: @nprizal your suggestion has a typo in `We introcuded a new`

Comment (nprizal) [reply README.md]: Yeah I noticed that, I've fixed it in my last commit

**Review by nprizal — APPROVED** [review]

**Pull request merged**